pkg/machines/libvirt.es6

```diff
@@ -620,4 +622,86 @@ function doUsagePolling (name, connectionName) {
     };
 }
 
+function handleEvent(dispatch, connectionName, line) {
+    // example lines, some with detail, one without:
+    // event 'reboot' for domain sid-lxde
+    // event 'lifecycle' for domain sid-lxde: Shutdown Finished
+    // event 'device-removed' for domain green: virtio-disk2
+    const eventRe = /event '([a-z-]+)' .* domain ([^:]+)(?:: (.*))?$/;
+
+    var match = eventRe.exec(line);
+    if (!match) {
+        console.warn("Unable to parse event, ignoring:", line);
+        return;
+    }
+    var [event_, name, info] = match.slice(1);
+
+    logDebug(`handleEvent(${connectionName}): domain ${name}: got event ${event_}; details: ${info}`);
+
+    // types and details: https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainEventID
+    switch (event_) {
+        case 'lifecycle':
+            let type = info.split(' ')[0];
+            switch (type) {
+                case 'Undefined':
+                    dispatch(undefineVm(connectionName, name));
+                    break;
+
+                case 'Defined':
+                    dispatch(updateOrAddVm({ connectionName, name }));
```

Comment:
Either "Defined" or "Started" or both need to call `getVM`.  Otherwise we never get all the data for a VM that is created while the page is open and listens for events.  Probably both.